provider/plugin/plugin.go

```diff
@@ -0,0 +1,337 @@
+		return nil, fmt.Errorf("failed to apply changes with code %d", resp.StatusCode)
+	}
+
+	b, err := io.ReadAll(resp.Body)
+	if err != nil {
+		recordsErrorsGauge.Inc()
+		log.Debugf("error reading response body: %s", err.Error())
+		return nil, err
+	}
+
+	endpoints := []*endpoint.Endpoint{}
+	err = json.Unmarshal(b, &endpoints)
+	if err != nil {
+		recordsErrorsGauge.Inc()
+		log.Debugf("error unmarshaling response body: %s", err.Error())
+		return nil, err
+	}
+	return endpoints, nil
+}
+
+// ApplyChanges will make a POST to remoteServerURL/records with the changes
+func (p PluginProvider) ApplyChanges(ctx context.Context, changes *plan.Changes) error {
+	u, err := url.JoinPath(p.remoteServerURL.String(), "records")
+	if err != nil {
+		applyChangesErrorsGauge.Inc()
+		log.Debugf("error joining path: %s", err.Error())
+		return err
+	}
+	b, err := json.Marshal(changes)
+	if err != nil {
+		applyChangesErrorsGauge.Inc()
+		log.Debugf("error marshaling changes: %s", err.Error())
+		return err
+	}
+
+	req, err := http.NewRequest("POST", u, bytes.NewBuffer(b))
+	if err != nil {
+		applyChangesErrorsGauge.Inc()
+		log.Debugf("error creating request: %s", err.Error())
+		return err
+	}
+	resp, err := p.client.Do(req)
+	if err != nil {
+		applyChangesErrorsGauge.Inc()
+		log.Debugf("error performing request: %s", err.Error())
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		applyChangesErrorsGauge.Inc()
+		log.Debugf("error from external provider, HTTP status code %d", resp.StatusCode)
+		return fmt.Errorf("failed to apply changes with code %d", resp.StatusCode)
+	}
+	return nil
+}
+
+// PropertyValuesEqual will call the provider doing a POST on `/propertyvaluesequal` which will return a boolean in the format
+// `{propertyvaluesequal: true}`
+// Errors in anything technically happening from the provider will return true so that no update is performed.
+// Errors will also be logged and exposed as metrics so that it is possible to alert on them if needed.
+func (p PluginProvider) PropertyValuesEqual(name string, previous string, current string) bool {
+	u, err := url.JoinPath(p.remoteServerURL.String(), "propertyvaluesequal")
+	if err != nil {
+		propertyValuesEqualErrorsGauge.Inc()
+		log.Debugf("error joining path: %s", err)
+		return true
+	}
+	b, err := json.Marshal(&PropertyValuesEqualRequest{
+		Name:     name,
+		Previous: previous,
+		Current:  current,
+	})
+	if err != nil {
+		propertyValuesEqualErrorsGauge.Inc()
+		log.Debugf("error marshaling request: %s", err)
+		return true
+	}
+
+	req, err := http.NewRequest("POST", u, bytes.NewBuffer(b))
+	if err != nil {
+		propertyValuesEqualErrorsGauge.Inc()
+		log.Debugf("error creating request: %s", err)
+		return true
+	}
+	resp, err := p.client.Do(req)
+	if err != nil {
+		propertyValuesEqualErrorsGauge.Inc()
+		log.Debugf("error performing request: %s", err)
+		return true
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
```

Comment:
I think the docs say StatusCode 20x is fine, but here it's only 200, so we should either change docs or what we expect here.

Reply:
Switched the docs to 200.